src/HealthChecks/Abstractions/src/HealthReportEntryDictionary.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Diagnostics.HealthChecks;
+/// <summary>
+/// Represents a collection of <see cref="HealthReportEntry"/> ordered by keys.
+/// </summary>
+public class HealthReportEntryDictionary : IReadOnlyDictionary<string, HealthReportEntry>
```

Comment:
This can pheraps be marked visible for only use in _DefaultHealthCheckService_?

Reply:
It can be marked as `internal` if it's not directly referenced in any other public API.